plugins/parsers/json_v2/parser.go

```diff
@@ -68,6 +71,13 @@ func (p *Parser) Init() error {
 }
 
 func (p *Parser) Parse(input []byte) ([]telegraf.Metric, error) {
+	reader := strings.NewReader(string(input))
+	body, _ := utfbom.Skip(reader)
+	input, err := io.ReadAll(body)
```

Comment:
Can we please make this an `internal` function? We trim BOMs in `inputs.file`, `inputs.tail`, `tools/custom_builder` and last but not least `config/config.go` (`trimBOM()`) and I guess we should use this new function in all those places (later).

Reply:
It would be useful to have this in telegraf shared code (like internal) but we could merge this first and factor it out into internal in a follow-up PR.